lib/iris/cube.py

```diff
@@ -2001,6 +2175,15 @@ def __getitem__(self, keys):
         for factory in self.aux_factories:
             cube.add_aux_factory(factory.updated(coord_mapping))
 
+        # slice the cell measures, if changed and add them to the cube
+        if self._cell_measures_and_dims != cube._cell_measures_and_dims:
```

Comment:
@marqh I don't see the point in this `if` statement, it may be left over from your previous implementation when cell measures were part of the cube metadata. Now, at this point in time, the new cube will always have an empty `_cell_measures_and_dims`.